core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -168,40 +169,48 @@ class ContainerPool(
             info(this, s"Shutting down: Not getting container for ${action.fullyQualifiedName} with ${auth.uuid}")
             None
         } else {
-            try {
-                val myPos = nextPosition.next()
-                info(this, s"""Getting container for ${action.fullyQualifiedName} of kind ${action.exec.kind} with ${auth.uuid}:
-                              | myPos = $myPos
-                              | completed = ${completedPosition.cur}
-                              | slack = ${slack()}
-                              | startingCounter = ${startingCounter.cur}""".stripMargin)
-                val key = ActionContainerId(auth.uuid, action.fullyQualifiedName, action.rev)
-                getImpl(1, myPos, key, () => makeWhiskContainer(action, auth)) map {
-                    case (c, initResult) =>
-                        val cacheMsg = if (!initResult.isDefined) "(Cache Hit)" else "(Cache Miss)"
-                        (c.asInstanceOf[WhiskContainer], initResult)
-                }
-            } finally {
-                completedPosition.next()
+            val key = ActionContainerId(auth.uuid, action.fullyQualifiedName, action.rev)
+            val myPos = nextPosition.next()
+            info(this, s"""Getting container for ${action.fullyQualifiedName} of kind ${action.exec.kind} with ${auth.uuid}:
+                          | myPos = $myPos
+                          | completed = ${completedPosition.cur}
+                          | slack = ${slack()}
+                          | startingCounter = ${startingCounter.cur}""".stripMargin)
+            val conResult = Try(getContainer(1, myPos, key, () => makeWhiskContainer(action, auth)))
+            completedPosition.next()
```

Comment:
i won't - the plan is 2 commits: semantic preserving refactor followed by init moving